### PR TITLE
Raftstore: reset the bucket after the regions merge

### PR DESCRIPTION
**Pull request by tonyxuqqi**
Signed-off-by: qi.xu <tonxuqi@outlook.com>

### What is changed and how it works?

Issue Number: Ref #12574

What's Changed:

```commit-message
Reset the buckets of target region after the merge, this is to refresh the buckets in-time. 
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore/src/store/fsm/peer.rs [hunk 1]
```diff
@@ -4118,6 +4118,8 @@ where
         source: metapb::Region,
     ) {
         self.register_split_region_check_tick();
+        self.fsm.peer.region_buckets = None;
```
Comment (BusyJay) [inline components/raftstore/src/store/fsm/peer.rs:4121]: How about moving it to L4167? We may need to check all other places that `size_diff_hint` is reset.

**Review by BusyJay — APPROVED** [review]

File: components/raftstore/src/store/fsm/peer.rs (hunk 1 above)
Comment (tonyxuqqi) [reply components/raftstore/src/store/fsm/peer.rs]: done.  Good point.

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**